corva/application.py

```diff
@@ -25,13 +30,71 @@ def __init__(self, middleware: Optional[List[Callable]] = None):
 
     def get_middleware_stack(
          self,
-         middleware: Optional[List[Callable]] = None
+         middleware: Optional[List[Callable]] = None,
+         tail_middleware: Optional[List[Callable]] = None
     ) -> List[Callable]:
         middleware = middleware or []
+        tail_middleware = tail_middleware or []
 
-        middleware_stack = middleware + self.user_middleware
+        middleware_stack = middleware + self.user_middleware + tail_middleware
 
         return middleware_stack
 
     def add_middleware(self, func: Callable) -> None:
         self.user_middleware.append(func)
+
+    def stream(
+         self,
+         func=None,
+         *,
```

Comment:
It would be better to pass parameters by name to not shuffle them. https://softwareengineering.stackexchange.com/a/384703

Reply:
In the same answer there are pitfalls:
> * It's yet another feature for new team members to learn when they're coming up to speed on Python and your codebase.
> * Extra complexity and distraction in the function definition. (What's that *?)
> * Some dev tools might not handle it well: syntax highlighters, diff tools, refactoring editors, code inspectors, C interop, RPC (remote procedure call) code generators, ...

We are implementing a user-facing library (an SDK). Because of that, I consider these pitfalls important enough to get rid of `*`. I'm biased toward engineering simplicity. You use your best judgment

Reply:
I will leave it for now for following reasons.

1. it can always be deleted, without affecting backward compatibility
2. it helps not to shuffle many arguments provided by the decorator (write `@stream(filter_by_timestamp=True` and `@stream(filter_by_depth=True)` vs `@stream(True, False)`, `@stream(False, True)`)
2. it is probably time for devs to learn about this feature as it was added in python 3.0

Reply:
I was wrong. It actually very useful here